### PR TITLE
Add new repo-token input

### PR DESCRIPTION
**Pull request by CalvinWilkinson**
<h1 style="font-weight:bold" align="center">Preview Feature Pull Request</h1>
<h2 style="font-weight:bold" align="center">✅Code Review Checklist✅</h2>

- [x] The **_head(source)_** branch for this pull request is a **_preview feature_** branch, with the correct naming syntax, in the following example:
  - A **_head(source)_** branch is the branch being merged into a target branch.
    <details closed><summary>Example</summary>

      ``` xml
      Syntax: preview/feature/<issue-num>-<description>
      Example: preview/feature/123-my-preview-feature
      ```
    </details>

- [x] The **_head(source)_** branch for this pull request is created from a **_preview release_** branch, with the correct naming syntax, in the following example:
  <details closed><summary>Example</summary>

    ``` xml
    Syntax: preview/v<major>.<minor>.<patch>-preview.<prev-num>
    Example: preview/v1.2.3-preview.4
    ```
  </details>

- [x] The **_base(target)_** branch for this pull request is a **_preview release_** branch, with the correct naming syntax, in the following example:
  - A **_base(target)_** branch is the branch that the **_head(source)_** branch is merging into.

    <details closed><summary>Example</summary>

      ``` xml
      Syntax: preview/v<major>.<minor>.<patch>-preview.<prev-num>
      Example: preview/v1.2.3-preview.4
      ```
    </details>

- [x] Pull request title matches the title of the associated issue.

- [x] Associated issue exists and is linked to this pull request.
  - One issue per pull request.

- [x] The labels attached to this PR match the labels attached to the associated issue.

- [x] My code follows the coding style of this project.
  - The style is enforced by the **_.editorconfig_** files in the project and displayed as warnings.  If there is an edge case, reach out and let's discuss it.

- [x] A https://github.com/KinsonDigital/VersionMiner/labels/preview label has been added to the pull request.

- [x] I have manually tested my code changes to the best of my ability.

- [x] All tests passed locally.
  - This is required because unit tests might pass locally but not in the CI environment during the status check process or vice-versa.
  - Tests might pass on the developer's machine but not necessarily on the code reviewer's machine.
  - This does not mean that the tests pass in both the local and CI environment.
  - Status checks run for this pull request when merging into the **_preview/v\*.\*.\*-preview.\*_** branch.  These status checks run every time a change is pushed to the pull request.  These checks validate version syntax, tagging, builds, unit tests, and more.

### CODE REVIEW

**Review by kselena — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -191,8 +194,9 @@ Result:
 - name: Get Version From C# Project File
     uses: KinsonDigital/VersionMiner@v1.0.0-preview.2
     with:
-        repo-owner: JohnDoe
+        repo-owner: JohnD
         repo-name: MyRepo
+        repo-token: ${{ secrets.GITHUB_TOKEN }}oe
```
Comment (KinsonDigitalAdmin) [inline README.md:199]: Fix the `repo-token` value:

```suggestion
        repo-token: ${{ secrets.GITHUB_TOKEN }}
```

File: VersionMiner/ActionInputs.cs [hunk 2]
```diff
@@ -27,6 +27,16 @@ public class ActionInputs
         HelpText = "Gets or sets the name of the repository.")]
     public string RepoName { get; set; } = string.Empty;
 
+    /// <summary>
+    /// Gets or sets the token used to authenticate to the repository owner set by the <see cref="RepoOwner"/> value
+    /// for a repository name set by the <see cref="RepoName"/> value.
```
Comment (KinsonDigitalAdmin) [inline VersionMiner/ActionInputs.cs:32]: Refactor code docs to below:

```suggestion
    /// Gets or sets the token used to authenticate to the repository set by <see cref="RepoName"/> which is owned by the <see cref="RepoOwner"/>.
```

File: VersionMiner/ActionInputs.cs [hunk 3]
```diff
@@ -27,6 +27,16 @@ public class ActionInputs
...
+    /// </summary>
+    [Option(
+        "repo-token",
+        Required = false,
+        HelpText = "The PAT used to authenticate to the repository.")]
```
Comment (KinsonDigitalAdmin) [inline VersionMiner/ActionInputs.cs:37]: Refactor help text:

```suggestion
        HelpText = "The GitHub or PAT token used to authenticate to the repository.")]
```

File: action.yml [hunk 4]
```diff
@@ -11,6 +11,10 @@ inputs:
   repo-name:
     description: 'The name of the repository'
     required: true
+  repo-token:
+    description: 'The PAT used to authenticate to the repository.'
```
Comment (KinsonDigitalAdmin) [inline action.yml:15]: Refactor description

```suggestion
    description: 'The GitHub or PAT token used to authenticate to the repository.'
```

File: Testing/VersionMinerTests/ActionInputsTests.cs [hunk 5]
```diff
@@ -34,6 +34,11 @@ public void Ctor_WhenConstructed_PropsHaveCorrectDefaultValuesAndDecoratedWithAt
         inputs.GetAttrFromProp<OptionAttribute>(nameof(ActionInputs.RepoName))
             .AssertOptionAttrProps("repo-name", true, "Gets or sets the name of the repository.");
 
+        inputs.RepoToken.Should().BeEmpty();
+        typeof(ActionInputs).GetProperty(nameof(ActionInputs.RepoToken)).Should().BeDecoratedWith<OptionAttribute>();
+        inputs.GetAttrFromProp<OptionAttribute>(nameof(ActionInputs.RepoToken))
+            .AssertOptionAttrProps("repo-token", false, "The PAT used to authenticate to the repository.");
```
Comment (KinsonDigitalAdmin) [inline Testing/VersionMinerTests/ActionInputsTests.cs:40]: Update the unit test to reflect the change in the `ActionInputs` class

```suggestion
            .AssertOptionAttrProps("repo-token", false, "The GitHub or PAT token used to authenticate to the repository.");
```

**Review by KinsonDigitalAdmin — CHANGES_REQUESTED** [review]

**Review by KinsonDigitalAdmin — DISMISSED** [review]

**Review by KinsonDigitalAdmin — APPROVED** [review]

File: Documentation/ReleaseNotes/Production-Release-Notes-TEMPLATE.md [hunk 6]
```diff
@@ -1,7 +1,8 @@
 <h1 align="center" style='color:mediumseagreen;font-weight:bold'>
-VersionMiner Release Notes - [add-prod-release-here]</h1>
+    VersionMiner Production Release Notes - [TODO: add-prod-release-here]
```
Comment (kselena) [inline Documentation/ReleaseNotes/Production-Release-Notes-TEMPLATE.md:2]: Do you need this TODO here? 
```suggestion
    VersionMiner Production Release Notes - [TODO: add-prod-release-here]
```

**Review by kselena — CHANGES_REQUESTED** [review]

File: Documentation/ReleaseNotes/Production-Release-Notes-TEMPLATE.md (hunk 6 above)
Comment (KinsonDigitalAdmin) [reply Documentation/ReleaseNotes/Production-Release-Notes-TEMPLATE.md:2]: The TODO is supposed to be in the release notes template.

**Review by kselena — APPROVED** [review]

**Pull request merged**